src/app/features/planner/dialog-schedule-task/dialog-schedule-task.component.ts

```diff
@@ -327,9 +329,14 @@ export class DialogScheduleTaskComponent implements AfterViewInit {
         this.selectedDate = tomorrow;
         break;
       case 2:
-        const nextMonday = tDate;
-        nextMonday.setDate(nextMonday.getDate() + ((1 + 7 - nextMonday.getDay()) % 7));
-        this.selectedDate = nextMonday;
+        const nextFirstDayOfWeek = tDate;
+        const dayOffset =
+          (this._dateAdapter.getFirstDayOfWeek() -
+            this._dateAdapter.getDayOfWeek(nextFirstDayOfWeek) +
+            7) %
+            7 || 7;
```

Comment:
The fallback deals with the case where the current day aligns with the `FIRST_DAY_OF_WEEK` setting (i.e. add a full week).